### PR TITLE
Improve README, Add NixOS Installation #9

### PR DESCRIPTION
**Pull request by tdro**
README Improvement for those who may want to install on NixOS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +161,54 @@ func = autoArgs: fn: args: ...
...
+Expose the package using the `configuration.nix` options,
+where `../program/nixdoc/default.nix` is the path to the above derivation named `default.nix`.
+Link the plugin file using `nix.extraOptions`.
+
+```nix
+{ pkgs, ... }:
+
+{
+  nix.extraOptions = ''
+    plugin-files = ${pkgs.callPackage../program/nixdoc/default.nix {}}/lib/libnix_doc_plugin.so
```
Comment (lf-) [inline README.md:204]: I believe there's a space missing after callPackage here.

File: README.md [hunk 2]
```diff
@@ -161,6 +161,54 @@ func = autoArgs: fn: args: ...
...
+
+  meta = with stdenv.lib; {
+    homepage = url;
+    license = licenses.lgpl3;
+    description = "A Nix documentation lookup tool that quickly dumps the docs of the library function";
+  };
+}
+```
+
+Expose the package using the `configuration.nix` options,
```
Comment (lf-) [inline README.md:195]: As a non NixOS user, I'm not sure where configuration.nix normally lives, which might be useful to add.

File: README.md (hunk 1 above)
Comment (tdro) [reply README.md]: Nice catch, thanks.

File: README.md (hunk 2 above)
Comment (tdro) [reply README.md]: Sure, tell me if the wording looks good.

**Comment by lf-** [comment]
Looks good. It's definitely up to stylistic preferences but you could change it to fetchFromGitHub which might (? I'm on mobile and Nixpkgs documentation is pretty much not usable because of it) download a tarball instead, which would have faster runtime.

**Comment by tdro** [comment]
> Looks good. It's definitely up to stylistic preferences but you could change it to fetchFromGitHub which might (? I'm on mobile and Nixpkgs documentation is pretty much not usable because of it) download a tarball instead, which would have faster runtime.

Yes you're right - I've switched the source fetch it to use `fetchFromGitHub`. I'm quite used to fetching none github repo's haha.

**Comment by lf-** [comment]
Merged. Thanks for the contribution!

**Pull request merged**